src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js

```diff
@@ -368,4 +368,35 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
         return this.jq.datePicker().data().primeDatePicker.viewDate;
     },
 
+    /**
+     * Sets the disabled dates.
+     * @param {string[] | Date[]} disabledDates The dates to disable.
+     */
+    setDisabledDates: function(disabledDates) {
+        this.jq.data().primeDatePicker.options.disabledDates = disabledDates;
+        this.updatePanel();
+    },
+
+    /**
+     * Sets the disabled days.
+     * @param {Number[]} disabledDays The days to disable.
+     */
+    setDisabledDays: function(disabledDays) {
+        this.jq.data().primeDatePicker.options.disabledDays = disabledDays;
+        this.updatePanel();
+    },
+
+    /**
+     * Update panel.
+     */
+    updatePanel: function() {
+        var pdp = this.jq.data().primeDatePicker;
+        if (pdp.options.disabledDates) {
+            for (var i = 0; i < pdp.options.disabledDates.length; i++) {
+                pdp.options.disabledDates[i] = pdp.parseOptionValue(pdp.options.disabledDates[i]);
+            }
+        }
```

Comment:
I tink this pdo.options for disabled dates has to get moved up to disabled Dates method no?